lib/help.js

```diff
@@ -43,81 +46,45 @@ class Help extends BaseCommand {
   }
 
   async help (args) {
-    const argv = this.npm.config.parsedArgv.cooked
+    // By default we search all of our man subdirectories, but if the user has
+    // asked for a specific one we limit the search to just there
+    let manSearch = 'man*'
+    if (/^\d+$/.test(args[0]))
+      manSearch = `man${args.shift()}`
 
-    let argnum = 0
-    if (args.length === 2 && ~~args[0])
-      argnum = ~~args.shift()
+    if (!args.length)
+      return this.npm.output(this.npm.usage)
 
     // npm help foo bar baz: search topics
-    if (args.length > 1 && args[0])
+    if (args.length > 1)
       return this.helpSearch(args)
 
-    const affordances = {
-      'find-dupes': 'dedupe',
-    }
-    let section = affordances[args[0]] || this.npm.deref(args[0]) || args[0]
-
-    // npm help <noargs>:  show basic usage
-    if (!section) {
-      npmUsage(this.npm, argv[0] === 'help')
-      return
-    }
-
-    // npm <command> -h: show command usage
-    if (this.npm.config.get('usage') &&
-      this.npm.commands[section] &&
-      this.npm.commands[section].usage) {
-      this.npm.config.set('loglevel', 'silent')
-      log.level = 'silent'
-      this.npm.output(this.npm.commands[section].usage)
-      return
-    }
+    let section = this.npm.deref(args[0]) || args[0]
 
-    let pref = [1, 5, 7]
```

Comment:
The old method would treat the number in the args as a suggestion, there is no need to search everywhere if we are explicitly asking for a given man number.  This was the source of most of the streamlining here.